entities/TradeService.js

```diff
@@ -0,0 +1,374 @@
+const { cloneDeep } = require('lodash');
+const PropertyManagementService = require('./PropertyManagementService');
+const WealthService = require('./WealthService');
+const Errors = require('./Errors');
+
+module.exports = class TradeService {
+  // returns { player, details: { askingFor: [], receiving: [] }} OR nothing?
+  static trade(UI, gameState) {
+    const { NEW, CANCEL, OFFER, ACCEPT } = TradeService.TradeStatus;
+
+    const players = gameState.players
+      .filter((p) => !p.bankrupt)
+      .map((p) => {
+        const {
+          tradeableProps,
+          untradeableProps,
+        } = PropertyManagementService.getPlayerProperties(gameState, p);
+        return {
+          ...p,
+          tradeableProps,
+          untradeableProps,
+        };
+      });
```

Comment:
This might be too much, but we could do:
```javascript
const getPlayerProperties = PropertyManagementService.getPlayerProperties.bind(PropertyManagementService, gameState);

const players = gameState.players
    .filter(p => !p.bankrupt)
    .map(p => [p, getPlayerProperties(p)])
    .map(([p, { tradeableProps, untradeableProps }]) => ({ ...p, tradeableProps, untradeableProps }));
```